__tests__/pages/webhooks/order-created.test.ts

```diff
@@ -0,0 +1,182 @@
+/** @jest-environment setup-polly-jest/jest-environment-node */
+
+import { PollyServer } from "@pollyjs/core";
+import * as joseModule from "jose";
+import { NextApiRequest, NextApiResponse } from "next";
+import toNextHandler from "../../../pages/api/webhooks/order-created";
+import { setupPollyMiddleware, setupRecording } from "../../pollySetup";
+import {
+  dummyOrderCreatedPayload,
+  mockRequest,
+  dummyFetchTaxesResponse,
+} from "../../utils";
+import Avatax from "ava-typescript";
+
+jest.mock("next/dist/compiled/raw-body/index.js", () => ({
+  __esModule: true,
+  default: jest.fn((_) => ({
+    toString: () => '{"dummy":12}',
+  })),
+}));
+
+const testDomain = "localhost:8000";
+describe("api/webhooks/order-created", () => {
+  const context = setupRecording();
+  beforeEach(() => {
+    process.env.SALEOR_DOMAIN = testDomain;
+    const server = context.polly.server;
+    setupPollyMiddleware(server as unknown as PollyServer);
+  });
+
+  it("rejects when saleor domain is missing", async () => {
+    const mockedCreateTransaction = jest
+      .spyOn(Avatax.prototype, "createTransaction")
+      .mockResolvedValue(dummyFetchTaxesResponse);
+
+    const domain = undefined;
+    const { req, res } = mockRequest({
+      method: "POST",
+      event: "order_created",
+      domain,
+    });
+
+    const orderPayload = dummyOrderCreatedPayload;
+    req.body = orderPayload;
+
+    await toNextHandler(
+      req as unknown as NextApiRequest,
+      res as unknown as NextApiResponse
```

Comment:
We could do this type assertion inside the `mockRequest` instead and this way we wouldn't have to write it each time we pass `req` and `res` to the handler function

Example: https://github.com/saleor/react-storefront/blob/main/apps/saleor-app-checkout/test-utils.ts#L24